pkg/operator/types.go

```diff
@@ -53,3 +53,25 @@ func MakeHostAliases(input []monitoringv1.HostAlias) []v1.HostAlias {
 
 	return output
 }
+
+func MakeK8sTopologySpreadConstraint(podLabels map[string]string, tsc monitoringv1.TopologySpreadConstraint) v1.TopologySpreadConstraint {
+	if len(tsc.AdditionalLabelSelectors) > 0 {
+		if tsc.LabelSelector == nil {
+			tsc.LabelSelector = &metav1.LabelSelector{
+				MatchLabels: make(map[string]string),
+			}
+		}
+
+		labelMap := map[monitoringv1.AdditionalLabelSelector]string{
```

Comment:
@simonpasquier due to [this](https://github.com/prometheus-operator/prometheus-operator/pull/5967#discussion_r1362156621) suggestion, we have to keep a map from the CRD value to Prometheus Operator Labels.

I'm not against the idea, I'm just wondering if is there a better place to have this map and avoid the object creation on every reconcile loop.

Reply:
@simonpasquier friendly ping 😄

Reply:
I wouldn't too worried about this but you can have a const instead.